kconfig-hardened-check.py

```diff
@@ -17,101 +17,123 @@
+    add_opt_check([Opt('STATIC_USERMODEHELPER',            'y', 'my', 'self_protection'), '']) # breaks systemd?
+
+    add_opt_check([Opt('SECURITY',                    'y', 'ubuntu18', 'security_policy'), ''])
+    add_opt_check([Opt('SECURITY_YAMA',               'y', 'ubuntu18', 'security_policy'), ''])
+    add_opt_check([Opt('SECURITY_SELINUX_DISABLE',    'is not set', 'ubuntu18', 'security_policy'), ''])
+
+    add_opt_check([Opt('ACPI_CUSTOM_METHOD',   'is not set', 'ubuntu18', 'cut_attack_surface'), ''])
+    add_opt_check([Opt('COMPAT_BRK',           'is not set', 'ubuntu18', 'cut_attack_surface'), ''])
+    add_opt_check([Opt('DEVKMEM',              'is not set', 'ubuntu18', 'cut_attack_surface'), ''])
+    add_opt_check([Opt('COMPAT_VDSO',          'is not set', 'ubuntu18', 'cut_attack_surface'), ''])
+
+    add_opt_check([Opt('LEGACY_VSYSCALL_NONE', 'y', 'kspp', 'cut_attack_surface'), '']) # 'vsyscall=none'
+    add_opt_check([Opt('BINFMT_MISC',          'is not set', 'kspp', 'cut_attack_surface'), ''])
+    add_opt_check([Opt('INET_DIAG',            'is not set', 'kspp', 'cut_attack_surface'), ''])
+    add_opt_check([Opt('KEXEC',                'is not set', 'kspp', 'cut_attack_surface'), ''])
+    add_opt_check([Opt('PROC_KCORE',           'is not set', 'kspp', 'cut_attack_surface'), ''])
+    add_opt_check([Opt('LEGACY_PTYS',          'is not set', 'kspp', 'cut_attack_surface'), ''])
+    add_opt_check([Opt('IA32_EMULATION',       'is not set', 'kspp', 'cut_attack_surface'), ''])
+    add_opt_check([Opt('X86_X32',              'is not set', 'kspp', 'cut_attack_surface'), ''])
+    add_opt_check([Opt('MODIFY_LDT_SYSCALL',   'is not set', 'kspp', 'cut_attack_surface'), ''])
+    add_opt_check([Opt('KEXEC_FILE',           'is not set', 'my', 'cut_attack_surface'), ''])
+    add_opt_check([Opt('LIVEPATCH',            'is not set', 'my', 'cut_attack_surface'), ''])
+    add_opt_check([Opt('USER_NS',              'is not set', 'my', 'cut_attack_surface'), '']) # user.max_user_namespaces=0
+    add_opt_check([Opt('IP_DCCP',              'is not set', 'my', 'cut_attack_surface'), ''])
+    add_opt_check([Opt('IP_SCTP',              'is not set', 'my', 'cut_attack_surface'), ''])
+    add_opt_check([Opt('FTRACE',               'is not set', 'my', 'cut_attack_surface'), ''])
+    add_opt_check([Opt('KPROBES',              'is not set', 'my', 'cut_attack_surface'), ''])
+    add_opt_check([Opt('PROFILING',            'is not set', 'my', 'cut_attack_surface'), ''])
+    add_opt_check([Opt('UPROBES',              'is not set', 'my', 'cut_attack_surface'), ''])
+    add_opt_check([Opt('BPF_JIT',              'is not set', 'my', 'cut_attack_surface'), ''])
+    add_opt_check([Opt('BPF_SYSCALL',          'is not set', 'my', 'cut_attack_surface'), ''])
+
+    add_opt_check([Opt('LKDTM',    'm', 'my', 'feature_test'), ''])
+
+
+def add_opt_check(check):
+    opt_checks[check[0].name] = check
```

Comment:
Um... Please correct me if I'm wrong. 
So we create the ordered dictionary 'opt_checks' with elements of this type:
'name' -> [ Opt('name_again', 'state', 'decision, 'reason'), 'check_result' ]

I guess we can make it better. 
1. Can we add the 'check_result' into the OptCheck?
2. Can we avoid the dictionary and use a list of OptCheck objects?

I'm stopping the review here until we work out the decision.

@anthraxx, thank you for your time and efforts, I appreciate them!

Reply:
1. Sure, why not... moving it there
2. just thought it allowes easy direct access to checks via config name, but as we don't use it other then in the debug mode i gonna replace it with a list. (we can still add a dict later if we really need it)

Reply:
ok, thank you!
Looking forward to your next version.
I'm going to introduce some minor fixes in master soon. You will have to do a trivial rebase.